packages/shared/yup-validation-resolver/useYupValidationResolver.ts

```diff
@@ -0,0 +1,49 @@
+import { useCallback } from 'react';
+import { Resolver } from 'react-hook-form';
+import { AnySchema, ValidationError } from 'yup';
+
+export type ValidationErrorMessage = { type: string; field?: string };
+
+export type ValidationErrorMessages = {
+  type?: string;
+  message?: string;
+  messages?: Record<string, ValidationErrorMessage>;
+};
+
+const useYupValidationResolver = <T>(validationSchema: AnySchema): Resolver =>
+  useCallback(
```

Comment:
no need for it to be a hook IMO... it can just be a normal JS function... u can use currying to pass on the  "validationSchema" to this function...
```suggestion
const yupValidationResolver = <T>(validationSchema: AnySchema): Resolver =>
```

Reply:
I actually disagree. This is used in the scope of a React component which depends on a schema object that depends on other hooks inside this scope. so even if it was a function rather than a hook it would still need to be wrapped inside a useCallback hook. having this as a hook means I can have this bit if implementation using useCallback only in one place (the hook) instead of having to repeat this implementation everywhere where I would consume this function.

Reply:
this comment is resolved... had a discussion over call and we are keeping it to be a hook with memoizing its props...